### ISSUE TITLE
Amp Stack - parameters controlled by check boxes are still active even when it is disabled

### ISSUE THREAD

**Issue by Gimmeapill** [brummer10/guitarix.vst#26]
Hi Hermann,

I found one minor issue while playing with the Neural Amp Modeler (which is super great by the way - thanks again!).

The Bass Boost, Presence, Reverb, LowEnd and AntiFizz are still activable with their respective check boxes even when the amp stack itself is disabled. 
It is easy to disable them but ideally that shouldn't be the case, as we want a signal as dry as possible for the full-rig models.

Cheers,

LX

**Comment by Gimmeapill** [comment]
Actually, nevermind: that is quite handy once we know about it.
Adding a bit of Presence/AntiFizz etc. without having the gain stage active make sense with the models.
Maybe just adding some visual indication as the behavior can be quite puzzling?

**Comment by brummer10** [comment]
Yes, I need to remove the checkbox to disable the amp stack, as it does in fact nothing. 
The main amp could only be disabled by selecting the "--" no tube option.

**Comment by Gimmeapill** [comment]
Thanks, I got it now.

I ran a few tests with the latest changes, and in Guitarix standalone everything works as expected. 
(It is really great to be able to remove the stuff I don't need and turn it into a "basic" NAM loader)

Now with the plugin, the UI still has the labels so it is still a bit confusing:
![Screenshot_2024-05-26_16-07-04](https://github.com/brummer10/guitarix.vst/assets/4774469/e8616949-0852-40ee-a8bb-342c1579a48b).

Also the latest changes (I think), break the restoration of the plugin state in Bitwig: "Could not read responses: End of stream".
After re-inserting the plugin instances and re-selecting the presets everything seems to be working again, but this might break things for other users...

**Comment by brummer10** [comment]
> Also the latest changes (I think), break the restoration of the plugin state in Bitwig: "Could not read responses: End of stream".

mm, that shouldn't happen. I need to check what's going on there. Normally this change should be backward compatible, as it only hide widgets when needed. I may add the labels to the hide list, was just a bit lazy.

**Comment by Gimmeapill** [comment]
I gave it a few more tries but I couldn't reproduce the issue again. I have now four instances opening on startup and presets are loading fine (also when loading/saving in Guitarix).
So far so good - thanks Hermann!

**Comment by brummer10** [comment]
That's good to know, thanks for testing and report back.

**Comment by kramlie** [comment]
I discovered one more bug related to this: The Master Gain is still applied even when hidden. The others have no effect though, as expected.

**Comment by brummer10** [comment]
Thanks for reporting, I need to check (and fix) that.

**Comment by brummer10** [comment]
I rechecked that and remember that's a feature, not a bug. Taking the Master Gain out of the chain could lead to unwanted volume jumps so we decided to keep it active.

**Comment by kramlie** [comment]
Sure, that makes sense. It's nice to have that extra control anyway. But then it should not be hidden when selecting the bypass tube.

**Comment by brummer10** [comment]
Done, The Master Gain control wont be hide any more when you switch to no tubes.

**Comment by kramlie** [comment]
Confirmed! Thanks, @brummer10!
